portal-impl/src/com/liferay/portal/service/impl/UserServiceImpl.java

```
@@ -3556,26 +3550,35 @@ protected long[] checkOrganizations(long userId, long[] organizationIds)
 					organizationIds = ArrayUtil.append(
 						organizationIds, organization.getOrganizationId());
 				}
-
-				oldOrganizationIds[i] = organization.getOrganizationId();
 			}
 		}
 
 		// Check that the administrator has the permission to add a new
 		// organization and that the organization membership is allowed
 
+		if (!PropsValues.ORGANIZATIONS_ASSIGNMENT_STRICT ||
+			permissionChecker.isCompanyAdmin()) {
+
+			return organizationIds;
+		}
+
+		boolean allowed = false;
+
 		for (long organizationId : organizationIds) {
-			if ((oldOrganizationIds != null) &&
-				ArrayUtil.contains(oldOrganizationIds, organizationId)) {
+			if (OrganizationPermissionUtil.contains(
+					permissionChecker, organizationId,
+					ActionKeys.MANAGE_USERS)) {
 
-				continue;
-			}
+				allowed = true;
 
-			Organization organization =
-				_organizationPersistence.findByPrimaryKey(organizationId);
+				break;
+			}
+		}
 
-			OrganizationPermissionUtil.check(
-				permissionChecker, organization, ActionKeys.ASSIGN_MEMBERS);
+		if (!allowed) {
```

and if you reach this point you can just throw the exception